src/codeu/chat/server/Server.java

```diff
@@ -99,7 +99,7 @@ public void onMessage(InputStream in, OutputStream out) throws IOException {
 
         Serializers.INTEGER.write(out, NetworkCode.NEW_USER_RESPONSE);
         Serializers.nullable(User.SERIALIZER).write(out, user);
-        logQueue.transactions.add("ADD-USER " + user.id.toString() + " " + "\""+ user.name + "\"" + " "+ user.creation.inMs());
+        logQueue.getTransactions().add("ADD-USER " + user.id.toString() + " " + "\""+ user.name + "\"" + " "+ user.creation.inMs());
```

Comment:
I think this change came out of some earlier comment I made about avoiding accessible fields (even package-private ones) and using accessor methods instead. For what it's worth, the new `.getTransactions()` version of this code looks more "right" to me than the old `.transactions` version.

There's something called the [Law of Demeter](https://en.wikipedia.org/wiki/Law_of_Demeter), which basically says "objects should know as little as possible about each other's internals". This code takes a big step in the direction of following that law.

Arguably, to *really* follow the Law of Demeter, you wouldn't expose the transactions list at all and would instead just have `logQueue.addTransaction()`. That way, if you ever changed the internals of `LogQueue` to no longer have a collection of transactions in it (maybe you stop buffering and all transactions are written immediately?), code that uses `LogQueue` would not change at all.

No need to actually do that here, but I still wanted to put in a plug for the Law of Demeter so that you'd think about it for next time.